crates/sui-core/src/authority/authority_store.rs

```diff
@@ -455,6 +474,14 @@ impl<const ALL_OBJ_VER: bool, S: Eq + Serialize + for<'de> Deserialize<'de>>
         Ok(())
     }
 
+    /// Insert a timestamp for a transaction digest.
+    /// Note this function is not thread/async safe when processing the same digest concurrently.
```

Comment:
another reason to move this stuff to IndexStore, since that data is updated only once.

(Also, I assume you meant that this function is not atomic when paired with a if-exists check - the function itself is thread safe)

Reply:
the function is not thread safe because race conditions may very well corrupt the states : https://en.wikipedia.org/wiki/Thread_safety

Reply:
No, its safe to access rocksdb concurrently from any number of threads. If it wasn't, the DBMap type wouldn't be marked Sync and your code wouldn't compile.

Reply:
(You're right that there is a race condition here, if you have multiple concurrent callers it's not defined which one will win, but it's still safe code. nothing can be corrupted by this call.)

Reply:
yeah "corruption" is not the right word. In database's jargons I should say this has no read-after-write consistency.

This code was moved in the new commit, so we don't need to struggle with the wordings any longer :)